scripts/services/add-localizations.js

```diff
@@ -0,0 +1,249 @@
+ */
+const getGroupedTranslations = async (localesFolder) => {
+    try {
+        // Get an array of locale directories in the base folder
+        const localesDirectories = await getDirNames(localesFolder);
+        const existingTranslations = [];
+        // Collect translations from each directory
+        localesDirectories.forEach((directory) => {
+            // File path to the translation file
+            const translationFilePath = path.join(localesFolder, directory, SERVICES_TRANSLATION_FILE);
+            // Check if the translation file exists
+            if (existsSync(translationFilePath)) {
+                // Read and parse the translation content
+                const translationContent = JSON.parse(readFileSync(translationFilePath));
+                // Group translations by id, locale and use (name, description etc.)
+                existingTranslations.push(groupFileContentByTranslations(translationContent, directory));
+            }
+        });
+        // Reduce the array of translations into a single grouped object
+        return existingTranslations.reduce((acc, obj) => {
+            // Merge translations for each component and locale
+            Object.entries(obj).forEach(([key, value]) => {
+                acc[key] = { ...acc[key], ...value };
+            });
+            return acc;
+        }, {});
+    } catch (error) {
+        logger.error(`Error getting grouped translations: ${error}`);
+        throw new Error(error);
+    }
+};
+
+/**
+ * @typedef {{groups: groupedFileObjects}} categoryLocalesTranslate
+ * @property {object} groupedFileObjects - An object containing grouped translations
+ * for a specific group within a category and locale.
+ */
+
+/**
+ * Asynchronously retrieves grouped translations for different locales based on specified directories.
+ *
+ * @param {string} localesFolder - The base path to the folder containing locale directories.
+ * @returns {categoryLocalesTranslate} A promise that resolves to an object representing grouped translations
+ * for different locales.
+ *
+ * @throws {Error} If there is an issue finding directories or retrieving grouped translations.
+ */
+const getLocales = async (localesFolder) => {
+    try {
+        // Initialize an object to store grouped translations
+        const categoryLocalesTranslate = {};
+        // Get grouped translations for all locale directories
+        categoryLocalesTranslate.groups = await getGroupedTranslations(localesFolder);
+        return categoryLocalesTranslate;
+    } catch (error) {
+        logger.error(`Error getting locales: ${error}`);
+        throw new Error(error);
+    }
+};
+
+/**
+ * Sorts an array of translation objects based on their keys.
+ *
+ * @param {Array<object>} translations - Array of translation objects.
+ * @returns {Array<object>} - Sorted array of translation objects.
+ */
+const sortByKeys = (translations) => translations.sort((a, b) => {
```

Comment:
sorting the keys of translations, maybe `sortTranslations`

Reply:
and why is it necessary to concatenate keys?

Reply:
This function checks if there are translations in base locale for groups from `services.json`, if there are no translations, then a key-value pair is added, where the key is `servicesgroup.${group.id}.name` , and the value is `TODO comment`. I get two arrays, one with existing translations and one with TODOs, which need to be merged to fill in the missing translations in the base locale. 
Perhaps this should be done as a separate command rather than within a general function